configuration/src/main/java/io/camunda/configuration/beanoverrides/BrokerBasedPropertiesOverride.java

```diff
@@ -795,112 +785,146 @@ private void populateCamundaExporter(final BrokerBasedProperties override) {
       exporter.setArgs(args);
     }
 
-    setArg(args, "history.retention.enabled", secondaryStorage.getRetention().isEnabled());
-    setArg(args, "history.retention.minimumAge", secondaryStorage.getRetention().getMinimumAge());
-
-    setArg(args, "connect.type", secondaryStorage.getType().name());
-    setArg(args, "connect.url", database.getUrl());
-    setArg(args, "connect.clusterName", database.getClusterName());
-    setArg(args, "connect.dateFormat", database.getDateFormat());
-    final var socketTimeout = database.getSocketTimeout();
-    setArg(args, "connect.socketTimeout", socketTimeout != null ? socketTimeout.toMillis() : null);
-    final var connectionTimeout = database.getConnectionTimeout();
-    setArg(
-        args,
-        "connect.connectTimeout",
-        connectionTimeout != null ? connectionTimeout.toMillis() : null);
+    exporter.setArgs(
+        ExporterConfiguration.of(io.camunda.exporter.config.ExporterConfiguration.class, args)
+            .apply(config -> applyCamundaExporterConfiguration(config, secondaryStorage, database))
+            .toArgs());
+  }
+
+  private void applyCamundaExporterConfiguration(
+      final io.camunda.exporter.config.ExporterConfiguration config,
+      final SecondaryStorage secondaryStorage,
+      final DocumentBasedSecondaryStorageDatabase database) {
 
-    // Add security configuration mapping
-    if (database.getSecurity() != null) {
-      setArg(args, "connect.security.enabled", database.getSecurity().isEnabled());
-      setArg(args, "connect.security.certificatePath", database.getSecurity().getCertificatePath());
-      setArg(args, "connect.security.verifyHostname", database.getSecurity().isVerifyHostname());
-      setArg(args, "connect.security.selfSigned", database.getSecurity().isSelfSigned());
+    applyCamundaExporterConfigurationRetention(
+        config.getHistory().getRetention(), secondaryStorage);
+    applyCamundaConfigurationConnect(config.getConnect(), secondaryStorage, database);
+    applyCamundaExporterConfigurationIndex(config.getIndex(), database);
+    applyCamundaExporterConfigurationHistory(config.getHistory(), database);
+    applyCamundaExporterConfigurationPostExportConfiguration(config.getPostExport(), database);
+    applyCamundaExporterConfigurationBulk(config.getBulk(), database);
+
+    if (database.getIncidentNotifier() != null) {
+      applyCamundaExporterConfigurationIncidentNotifier(config.getNotifier(), database);
     }
 
-    populateInterceptorPlugins(database, args);
+    config.setCreateSchema(database.isCreateSchema());
```

Comment:
🔧 the remaining configurations below feel a bit arbitrary, could they be grouped? or at least also moved into an additional helper (even if it's call applyMiscConfigurations) - that would keep this method on the same level of detail and easy to scan.